homeassistant/components/octoprint/sensor.py

```diff
@@ -124,11 +151,26 @@ def __init__(
     @property
     def native_value(self):
         """Return sensor state."""
+
+        # Get printer data from the coordinator
         printer: OctoprintPrinterInfo = self.coordinator.data["printer"]
         if not printer:
             return None
 
-        return printer.state.text
+        # Attempt to translate the state string from the API into an internal state value
+        api_state = printer.state.text
+        if value := _API_STATE_VALUE.get(api_state):
+            return value
+
+        # Got an unexpected state string from the API; fabricate a new internal state value and save it for next time
```

Comment:
A warning would probably be useful here, so that we know that we should add a new value in our code.